### PR TITLE
feat: Add `CaseInsensitiveSet` with original casing support

### LINKED ISSUE #515: bug: Registries are case sensitive
### Describe the issue

Currently registries are case sensitive in the way they are handled in the CLI. Mentioned in #512.  

### Registry Link

https://github/ieedan/std

### Reproduction Link

https://github/ieedan/std

### Steps to reproduce

If you were to run:

```sh
jsrepo init github/ieedan/std

# then

jsrepo init github/Ieedan/std
```

This would add both registries to the config.

### Validations

- [x] I have checked other issues to see if my issue was already reported or addressed

### PR DESCRIPTION
**Pull request by anishshobithps**
- [X] Adds a custom `CaseInsensitiveSet` class using a `Map` for case-insensitive operations.
- [X] Preserves first-in original casing while ensuring deduplication.
- [X] Implements `add`, `delete`, `has`, `clear`, `forEach`, `entries`, `keys`, `values`, and iteration.
- [X] Locale-aware normalization via `toLocaleLowerCase()`.
- [X] Includes comprehensive test suite for common & edge cases.
  
Replaces `Set` with `CaseInsensitiveSet` in `init` and `add`.

Closes #515

### CODE REVIEW

File: tests/case-insensitive-set.test.ts [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+			});
+
+			expect(values).toEqual(['A', 'B']);
+		});
+
+		it('respects thisArg parameter', () => {
+			const set = new CaseInsensitiveSet<string>(['test']);
+			const context = { prefix: 'Hello: ' };
+
+			// biome-ignore lint/complexity/noForEach: <explanation>
```
[the remark this answers is not in the document]
Comment (anishshobithps) [reply tests/case-insensitive-set.test.ts]: [3bdef15](https://github.com/jsrepojs/jsrepo/pull/576/commits/3bdef15d596082d50458446a4541c436cec5779e)

File: tests/case-insensitive-set.test.ts [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+	beforeAll(() => {
+		if (fs.existsSync(testDir)) {
+			fs.rmSync(testDir, { recursive: true });
+		}
+		fs.mkdirSync(testDir, { recursive: true });
+		process.chdir(testDir);
+	});
+
+	afterAll(() => {
+		process.chdir(__dirname);
```
[the remark this answers is not in the document]
Comment (anishshobithps) [reply tests/case-insensitive-set.test.ts]: [1435a78](https://github.com/jsrepojs/jsrepo/pull/576/commits/1435a78d416ece1add517373602a7486d4f5c501)
